security/fma-isthmus-l2-pectra.md

```diff
@@ -0,0 +1,207 @@
+    
+- **Detection:** Alert for L2 Unsafe liveness should be triggered
+- **Recovery Path(s)**: This would require an emergency update if it occurred fixing the network upgrade transaction in rollup node client software (like `op-node`).
+
+### FM2: Chain halt due to request hash mismatch
+
+- **Description:** Requests hash was added to the block header. If it mismatches between the EL client and the rollup node, the chain could halt.
+- **Risk Assessment:** High severity, low likelihood
+- **Mitigations:**
+  1. End-to-end test ensuring no requests are included in blocks
+      - https://github.com/ethereum-optimism/optimism/pull/14253
+  2. Simplicity of validation rules - always must be an empty hash
+- **Detection:** Alert for L2 Unsafe liveness should be triggered
+- **Recovery Path(s)**: This would require an emergency update if it occurred fixing requests hash logic in respective clients (e.g. `op-node`, `op-geth`, `op-reth`, ...)
+
+### FM3: EIP-7702 transactions cannot be included on the chain
+
+- **Description:** A client implementation could have a bug that disallows SetCode (EIP-7702) transactions from being included on chain. This could cause an unintended fork by the affected client.
+- **Risk Assessment:** High impact, low likelihood
+- **Mitigations:** 
+	1. End-to-end tests of set code transactions before and after the fork
+        - https://github.com/ethereum-optimism/optimism/pull/14197
+        - https://github.com/ethereum-optimism/optimism/pull/14288
+- **Detection:** Sending set code transactions would fail - manual detection is most likely.
+- **Recovery Path(s)**: This would require an emergency client update of the implementation with the bug (e.g. `op-reth`, `op-geth`, ...).
+
+### FM4: BLS Precompiles could cause increased FP Program execution time
+
+- **Description:** If implemented in the FP program, BLS precompiles could greatly increase computation time.
+- **Risk Assessment:** High impact, low likelihood
+- **Mitigations:**
+	1. Fully mitigated by using an accelerated precompile that calls out to L1 instead of calculating inside the program.
+        - [Specs](https://github.com/ethereum-optimism/specs/blob/b5e0fa98881171f658f782597a46b641e8f3dfd0/specs/protocol/isthmus/exec-engine.md#evm-changes)
+- **Detection:** op-program trace runner computation time would increase significantly (should trigger an alert)
+- **Recovery Path(s)**: If this occurred, which it can't with the mitigations, we'd have to make op-program efficient enough to execute
+BLS precompiles in sufficient time.
+
+### FM5: BLS Precompiles could fail to execute in the FP program
+
+- **Description:** If implemented in the FP program, BLS precompiles could fail to execute. This could cause certain blocks to be unprovable.
+- **Risk Assessment:** High impact, low likelihood
+- **Mitigations:**
+	1. Fully mitigated by using an accelerated precompile that calls out to L1 instead of calculating inside the program. This means it should 
```

Comment:
Where did we tests that the E2E  tests for transaction using BLS producing block and are submitted as expected by the batcher correctly?

Reply:
We check geth here: https://github.com/ethereum-optimism/optimism/commit/e76409e18ee82b4765ae9d51991ffeae1e032671#diff-b708c9a61d73e393273eb3a63bf8538ab51214cd3d33fa2aff74588eddb45c2cR1101-R1103

But, I think we should also check that op-program validates a block correctly with BLS precompiles. I'll add that test and link it here.

Reply:
@meyer9 I don't see this test here?